js/lib/src/state/namespace.ts

```diff
@@ -17,6 +17,108 @@ export enum Permissiveness {
   Namespace,
 }
 
+// conver enum to an anchor compatible type
+export function convertPermissiveness(p: Permissiveness): {} {
+  switch (p) {
+    case Permissiveness.All:
+      return { all: {} };
+    case Permissiveness.Whitelist:
+      return { whitelist: {} };
+    case Permissiveness.Blacklist:
+      return { blacklist: {} };
+    case Permissiveness.Namespace:
+      return { namespace: {} };
+  }
```

Comment:
Do we want to throw if an invalid permissiveness is passed in?

Reply:
I dont think `AnchorPermissivenessType` is exactly correct, I need different enum for this one. https://github.com/raindrops-protocol/raindrops/blob/4f1fd9c8847ae2f6da5a45a2ea6a743d084b2195/js/lib/src/state/common.ts#L50